### PR TITLE
feat(FEC-10387): add support for referenceId filter in provider

### PR DESCRIPTION
**Pull request by RoyBregman**
### Description of the Changes

- Added to MediaEntryLoader a new property - referenceId and allowed it אם be valid also with referenceId

- Passed the referenceId all the way from provider to the BE Service

- Fixed meta data service to use the entry id that comes from the service callback instead of the entryId from provider load  because when calling multiRequest with the referenceId we don't have the entryId yet.

Solves FEC-10387

### CheckLists

- [X] changes have been done against master branch, and PR does not conflict
- [X] new unit / functional tests have been added (whenever applicable)
- [X] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [X] Docs have been updated

### CODE REVIEW

File: test/src/k-provider/ovp/provider.spec.js [hunk 1]
```diff
@@ -83,6 +83,28 @@ describe('OVPProvider.partnerId:1082342', function () {
...
+    provider = new OVPProvider({partnerId: partnerId, uiConfId: 38621471}, playerVersion);
+    sinon.stub(MultiRequestBuilder.prototype, 'execute').callsFake(function () {
+      return new Promise(resolve => {
+        resolve({response: new MultiRequestResult(BE_DATA.EntryWithUIConfNoDrmData.response)});
+      });
+    });
+    provider.getMediaConfig({referenceId: '1_rsrdfext'}).then(
+      mediaConfig => {
+        try {
+          mediaConfig.should.deep.equal(MEDIA_CONFIG_DATA.WithPluginsNoDrm);
```
Comment (yairans) [inline test/src/k-provider/ovp/provider.spec.js:96]: What does this test check?

**Review by yairans — CHANGES_REQUESTED** [review]

File: flow-typed/types/media-info.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 // @flow
 declare type OVPProviderMediaInfoObject = {
-  entryId: string,
-  ks?: string
+  entryId?: string,
+  ks?: string,
+  referenceId?: string,
```
[the remark this answers is not in the document]
Comment (dan-ziv) [reply flow-typed/types/media-info.js:5]: referenceId and entryId cannot be shown together as optional, one of them is always mandatory, so I would suggest:
```js
declare type OVPProviderMediaInfoObject =
  | {
      entryId: string,
      ks?: string
    }
  | {
      referenceId: string,
      ks?: string
    };
```

[the remark this answers is not in the document]
Comment (dan-ziv) [reply flow-typed/types/media-info.js:5]: or 
```js
declare type OVPProviderMediaInfoObject = (
  | {
      entryId: string
    }
  | {
      referenceId: string
    }
) & {
  ks?: string
};
```

File: test/src/k-provider/ovp/provider.spec.js (hunk 1 above)
Comment (RoyBregman) [reply test/src/k-provider/ovp/provider.spec.js:96]: That it is valid. before the changes it would not pass

Comment (RoyBregman) [reply test/src/k-provider/ovp/provider.spec.js:96]: I agree its not much of a test... Do have any other test I can add?

File: src/k-provider/ovp/loaders/media-entry-loader.js [hunk 3]
```diff
@@ -75,6 +79,6 @@ export default class OVPMediaEntryLoader implements ILoader {
    * @returns {boolean} Is valid
    */
   isValid(): boolean {
-    return !!this._entryId;
+    return !!this._entryId || !!this._referenceId;
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/k-provider/ovp/loaders/media-entry-loader.js:82]: This is not same

File: test/src/k-provider/ovp/provider.spec.js (hunk 1 above)
Comment (yairans) [reply test/src/k-provider/ovp/provider.spec.js:96]: I would check the referenceId passed on the request

Comment (RoyBregman) [reply test/src/k-provider/ovp/provider.spec.js]: done

Comment (OrenMe) [reply test/src/k-provider/ovp/provider.spec.js]: @yairans please re-review

**Review by OrenMe — COMMENTED** [review]
@RoyBregman please fix PR(travis and conflict)

File: src/k-provider/ovp/loaders/entry-list-loader.js [hunk 4]
```diff
@@ -53,7 +53,15 @@ export default class OVPEntryListLoader implements ILoader {
     const config = OVPConfiguration.get();
     const requests: Array<RequestBuilder> = [];
     params.entries.forEach(entry => {
-      requests.push(OVPBaseEntryService.list(config.serviceUrl, params.ks, entry.entryId || entry, params.redirectFromEntryId));
+      requests.push(
+        OVPBaseEntryService.list(
+          config.serviceUrl,
+          params.ks,
+          entry.entryId || (typeof entry === 'string' && entry),
```
Comment (yairans) [inline src/k-provider/ovp/loaders/entry-list-loader.js:60]: this means you pass `false` when `typeof entry !== 'string'`. while you have to pass `string`

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/k-provider/ovp/loaders/entry-list-loader.js (hunk 4 above)
Comment (RoyBregman) [reply src/k-provider/ovp/loaders/entry-list-loader.js:60]: No, either entry is object with entryId prop or entry is a string

Comment (yairans) [reply src/k-provider/ovp/loaders/entry-list-loader.js:60]: so you can remove the `typeof` check

**Review by yairans — DISMISSED** [review]

File: flow-typed/types/media-info.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (OrenMe) [reply flow-typed/types/media-info.js:4]: @RoyBregman maybe I missed context but why was this dismissed?

[the remark this answers is not in the document]
Comment (RoyBregman) [reply flow-typed/types/media-info.js:4]: Wasn't possible. Tried different options and couldn't make it work

[the remark this answers is not in the document]
Comment (RoyBregman) [reply flow-typed/types/media-info.js:4]: I consulted with Dan and together decided to let it go

[the remark this answers is not in the document]
Comment (OrenMe) [reply flow-typed/types/media-info.js:4]: It is possible, see example [here](https://flow.org/try/#0C4TwDgpgBBB2wCcQEkAmAVc0C8UDeAPgFBSllyIqoBcUAzogJawDmANCWaQNZ0D8tBgmYtOpAgF8ioSFAQQAZhHmwAxhDSZZuQmLLylK9WkFNWHLj36nhrPZIDc0rFADyANQAKnhAHsAboyoygCyEKiMAIbIsAq+rgBGAFYQqsBQuAAUFEiaLgRyispwxhhYAJRORESqvrAMMPC5qADivr40bl4+AUGh4VExcYkpaRn4OVS0AESRdKhz02xQvDMAjABMAMwALNMSVbX16QZobR20Ht5+gcEIYRHRsfHJqek6BsVqGp2z84vLVZQaabXb7Q51BoJSKdK49W79R5DF6jd4TJpTYFzBZ0JaFQwlH4zbEAlZ0dbbPYHIhAA)

File: test/src/k-provider/ovp/provider.spec.js [hunk 5]
```diff
@@ -83,6 +83,23 @@ describe('OVPProvider.partnerId:1082342', function () {
...
 
+  it('should be able to get media config by referenceId', done => {
+    provider = new OVPProvider({partnerId: partnerId, uiConfId: 38621471}, playerVersion);
+    sinon.stub(MultiRequestBuilder.prototype, 'execute').callsFake(function () {
+      this.requests[1].params.filter.referenceIdEqual.should.equal('1_rsrdfext');
+      done();
+      return new Promise(() => {});
+    });
+    provider.getMediaConfig({referenceId: '1_rsrdfext'}).then(mediaConfig => {
+      try {
```
Comment (yairans) [inline test/src/k-provider/ovp/provider.spec.js:94]: this is unreachable code. the test already done in the `execute` stub. move the `try catch` there

**Review by yairans — CHANGES_REQUESTED** [review]

File: flow-typed/types/media-info.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (RoyBregman) [reply flow-typed/types/media-info.js:4]: So as we discussed it is possible but require going over the entire code and adjusting the types.
Closing the conversation

File: src/k-provider/ovp/loaders/entry-list-loader.js (hunk 4 above)
Comment (RoyBregman) [reply src/k-provider/ovp/loaders/entry-list-loader.js]: Yep, it also fixed flow

File: test/src/k-provider/ovp/provider.spec.js (hunk 5 above)
Comment (RoyBregman) [reply test/src/k-provider/ovp/provider.spec.js:94]: Please check now

**Review by yairans — APPROVED** [review]

**Pull request merged**